packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/migration.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { PublicMethodsOf } from '@kbn/utility-types';
+import type { SavedObjectUnsanitizedDoc } from '@kbn/core-saved-objects-server';
+import type { IKibanaMigrator } from '@kbn/core-saved-objects-base-server-internal';
+
+export type IMigrationHelper = PublicMethodsOf<MigrationHelper>;
+
+/**
+ * Helper for document migrations.
+ */
+export class MigrationHelper {
+  private migrator: IKibanaMigrator;
+
+  constructor({ migrator }: { migrator: IKibanaMigrator }) {
+    this.migrator = migrator;
+  }
+
+  /**
+   * Migrate the given SO document, throwing if a downgrade if required.
+   * This function is meant to be used by write-access APIs (create...) for documents provided as input.
+   * before storing it in the index. It will therefor throw if the document is in a higher / unknown version.
+   */
```

Comment:
```suggestion
   * Migrate the given SO document, throwing if a downgrade is required.
   * This function is meant to be used by write APIs (create, update) for documents provided as input.
   * before storing it in the index. It will therefore throw if the document is in a higher / unknown version.
   */
```